### ISSUE TITLE
map_collect_result_unit does not recognize for loop

### ISSUE THREAD

**Issue by matthiasbeyer** [rust-lang/rust-clippy#7710]
### What it does

This is a suggestion for the `map_collect_result_unit` lint.
I just refactored this piece of code:

```rust
fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
    for (uuid, err) in self.0.iter() {
        writeln!(f, "{}: {}", uuid, err)?;
    }
    Ok(())
}
```

into this:

```rust
fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
    self.0.iter().map(|(uuid, err)| writeln!(f, "{}: {}", uuid, err)).collect()
}
```

And then the lint told me that I can do

```rust
fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
    self.0.iter().try_for_each(|(uuid, err)| writeln!(f, "{}: {}", uuid, err))
}
```

But maybe it is possible that the lint also tells me that my first version can be written better?

I don't know whether this is possible, but it would be nice to catch the for-loop-version as well here.

### Categories (optional)

- Kind: `clippy::style`

**Comment by camsteffen** [comment]
The for loop is idiomatic so we should not lint it with the allow-by-default lint. In this case it saved 3 lines of code, but that is not always the case.
